Back-porting clarity tabs keyboard navigation to v1

bashibozuk opened the pull request:
- cherry picked 67a5512d7b0c353d003a84fe2bfc0010b5a8e9c5
- New key focus directive
- Accessible tabs + overflow; using the key focus directive as a basis

Signed-off-by: Vasil Atanasov <vasila@vmware.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [x] Other... Please describe: Backport

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: 4187

## What is the new behavior?

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

Jinnie commented:
Missing Circle CI configuration for V1. We can probably merge it without it.

Jinnie commented on gemini/tests/set4/tabs.js:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
```
We can revert this. As there are no code changes in the file, this will be only commit noise.

Jinnie commented on src/clr-angular/layout/tabs/_tabs.clarity.scss:
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+// Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
```
nit: 2020

Jinnie commented on src/clr-angular/layout/tabs/tabs.module.ts:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
```
nit: 2020

Jinnie commented on src/clr-angular/layout/tabs/tabs.ts:
```diff
@@ -9,34 +9,43 @@ import {
-                <ng-container *ngIf="link.tabsId === tabsId && !link.inOverflow"
-                              [ngTemplateOutlet]="link.templateRefContainer.template">
+                <ng-container *ngIf="link.tabsId === tabsId && !link.inOverflow">
+                    <li role="presentation" class="nav-item">
```
Probably this `nav-item` class may be responsible for the tab-width changes we see in the gemini screenshots.
I'd like to revert this resizing for V1 as it would be a visual breaking change that may lead to problems in the layouts of products that update to the next V1.

Jinnie reviewed (commented):
These files should go away. They are related to the vertical layout functionality, which we didn't want to port:
```
gemini/screens/tabs/angular/vertical-hover/chrome.png
gemini/dark-screens/tabs/angular/vertical-hover/chrome.png
```

vasilsatanasov committed:
Back-porting clarity tabs keyboard navigation to v1
- cherry picked 67a5512d7b0c353d003a84fe2bfc0010b5a8e9c5
- New key focus directive
- Accessible tabs + overflow; using the key focus directive as a basis

Signed-off-by: Vasil Atanasov <vsatanasov@gmail.com>

Jinnie reviewed (approved):
The backport seems legit to me. The additional outline in the gemini screenshot is expected, due to the popup auto focus.
I'd appreciate one more review before merging, as it wasn't just a cherry-pick, but needed manual separation of the tab-keys from vertical-tabs code.

bdryanovski commented on src/clr-angular/layout/tabs/tabs.ts:
```diff
@@ -88,12 +108,60 @@ export class ClrTabs implements AfterContentInit {
+
+  inOverflow() {
+    return (
+      this.tabLinkElements.indexOf(document.activeElement as HTMLElement) > -1 &&
```
This could be simplified `this.tabLinkElements.includes(document.activeElement)`

Jinnie replied:
It is part of the original back-ported code. Actually, it can be considered as a note to me :) 
I remember I did it like this, because of questionable support of array.includes in the IE browser/s, though we later agreed that this gets handled gently at compilation time.
